### PR TITLE
feat(docs): Implement new component overview

### PR DESCRIPTION
**Pull request by MajaZarkova**
Relates to #2298 

- Remove due indicator
- Replace icon buttons with icons
- Use new icons

## Checklist

- [x] All changes are documented in the documentation app (folder `apps/docs`)

### CODE REVIEW

File: apps/docs/src/.vitepress/components/ComponentRoadmap.vue [hunk 1]
```diff
@@ -39,7 +34,7 @@ const usedStatus = computed(() => {
  * Components that should be displayed (considers the show all button).
  */
 const displayedComponents = computed(() => {
-  return sortedComponents.value.slice(0, showAll.value ? undefined : 24);
+  return sortedComponents.value.slice(0, showAll.value ? undefined : 16);
```
Comment (larsrickert) [inline apps/docs/src/.vitepress/components/ComponentRoadmap.vue:37]: Can we change the default value of `showAll` to `true` so the user sees all components initially and can then clic to show less? I already talked with Martin about this and its okay for him

File: apps/docs/src/index.data.ts
Comment (larsrickert) [inline apps/docs/src/index.data.ts:1]: Did you check the components and their statuses with the ones mentioned in #2298?
Also the data grid exists twice :)

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: apps/docs/src/index.data.ts
Comment (larsrickert) [inline apps/docs/src/index.data.ts:1]: I discussed a few status changes with Martin:
- mark button and toast as "implemented"
- add "Notification center" as "planned"

I also updated the ticket

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: apps/docs/src/index.data.ts [hunk 2]
```diff
@@ -63,168 +63,145 @@ export default defineLoader({
...
     };
 
-    const basicComponentsDueDate = "Q2/2024";
-
     const components: HomePageData["components"] = [
       {
         name: "Button",
-        dueDate: basicComponentsDueDate,
-        status: "implemented",
+        status: "in-progress",
```
[the remark this answers is not in the document]
Comment (MajaZarkova) [reply apps/docs/src/index.data.ts:69]: @larsrickert  According to the ticket some of the components does not have a storybook file yet, but are market 'in-progress'. Maybe we can discuss this on the dev exchange.

**Pull request merged**